File: src/main/java/com/github/trcdeveloppers/clayium/common/GuiHandler.java
package com.github.trcdeveloppers.clayium.common;

import com.github.trcdeveloppers.clayium.common.blocks.machines.clayworktable.ClayWorktableContainer;
import com.github.trcdeveloppers.clayium.common.blocks.machines.clayworktable.TileClayWorkTable;
import com.github.trcdeveloppers.clayium.client.gui.GuiClayWorkTable;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

import javax.annotation.Nullable;

public class GuiHandler implements IGuiHandler {
    public static final int CLAY_WORK_TABLE = 1;

    @Nullable
    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));
        if (te == null) { return null; }

        switch (ID) {
            case CLAY_WORK_TABLE:
                return new ClayWorktableContainer(player.inventory, (TileClayWorkTable) te);
        }
        return null;
    }

    @Nullable
    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));
        if (te == null) { return null; }

        switch (ID) {
            case CLAY_WORK_TABLE:
                return new GuiClayWorkTable(player.inventory, (TileClayWorkTable) te);
        }
        return null;
    }
}

File: src/main/kotlin/com/github/trcdeveloppers/clayium/common/GuiHandler.kt
package com.github.trcdeveloppers.clayium.common

import com.github.trcdeveloppers.clayium.client.gui.GuiClayWorkTable
import com.github.trcdeveloppers.clayium.common.blocks.machine.clayworktable.ClayWorktableContainer
import com.github.trcdeveloppers.clayium.common.blocks.machine.clayworktable.TileClayWorkTable
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

class GuiHandler : IGuiHandler {
    companion object {
        const val CLAY_WORK_TABLE = 1
    }

    override fun getServerGuiElement(id: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        val te = world.getTileEntity(BlockPos(x, y, z)) ?: return null
        when (id) {
            CLAY_WORK_TABLE -> return ClayWorktableContainer(player.inventory, te as TileClayWorkTable)
        }
        return null
    }

    override fun getClientGuiElement(id: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        val te = world.getTileEntity(BlockPos(x, y, z)) ?: return null
        when (id) {
            CLAY_WORK_TABLE -> return GuiClayWorkTable(player.inventory, te as TileClayWorkTable)
        }
        return null
    }
}
